File: android-app/app/src/main/java/com/scamshield/ai/repository/AuthRepository.kt
package com.scamshield.ai.repository

import android.content.Context
import com.scamshield.ai.api.RetrofitClient

class AuthRepository(context: Context) {
    private val api = RetrofitClient.getApiService(context)

    suspend fun login(email: String, pass: String) = api.login(mapOf("email" to email, "password" to pass))
    
    suspend fun register(name: String, email: String, pass: String) = 
        api.register(mapOf("name" to name, "email" to email, "password" to pass))

    suspend fun verifyRegistration(email: String, otp: String) = 
        api.verifyRegistration(mapOf("email" to email, "otp" to otp))
    
    suspend fun getProfile() = api.getProfile()

    suspend fun forgotPassword(email: String) = api.forgotPassword(mapOf("email" to email))

    suspend fun resetPassword(email: String, otp: String, pass: String) = 
        api.resetPassword(mapOf("email" to email, "otp" to otp, "password" to pass))

    suspend fun logout() = try { api.logout() } catch (e: Exception) { null }

    suspend fun getAdminBlacklist() = api.getAdminBlacklist()
    suspend fun addAdminBlacklist(data: String, type: String, reason: String) = 
        api.addAdminBlacklist(com.scamshield.ai.model.AddBlacklistRequest(data, type, reason))
    suspend fun deleteAdminBlacklist(id: Int) = api.deleteAdminBlacklist(id)
    suspend fun getAdminReports() = api.getAdminReports()
    suspend fun approveAdminReport(id: Int) = api.approveAdminReport(id)
    suspend fun rejectAdminReport(id: Int) = api.rejectAdminReport(id)
}

File: flutter_app/lib/repositories/auth_repository.dart
import 'dart:convert';
import '../core/network/api_client.dart';
import '../core/storage/secure_storage.dart';
import '../models/auth_response.dart';
import '../models/user.dart';

class AuthRepository {
  final ApiClient apiClient;
  final SecureStorageService storageService;

  AuthRepository({required this.apiClient, required this.storageService});

  Future<AuthResponse> login(String email, String password) async {
    try {
      final response = await apiClient.dio.post(
        '/api/v1/auth/login',
        data: {'email': email, 'password': password},
      );
      final authResponse = AuthResponse.fromJson(response.data);
      if (authResponse.token != null) {
        await storageService.saveToken(authResponse.token!);
      }
      if (authResponse.user != null) {
        await storageService.saveUserData(jsonEncode(authResponse.user!.toJson()));
      }
      return authResponse;
    } catch (e) {
      throw ApiClient.getErrorMessage(e);
    }
  }

  Future<AuthResponse> register(String name, String email, String password) async {
    try {
      final response = await apiClient.dio.post(
        '/api/v1/auth/register',
        data: {'name': name, 'email': email, 'password': password},
      );
      return AuthResponse.fromJson(response.data);
    } catch (e) {
      throw ApiClient.getErrorMessage(e);
    }
  }

  Future<AuthResponse> verifyRegistration(String email, String otp) async {
    try {
      final response = await apiClient.dio.post(
        '/api/v1/auth/verify-registration',
        data: {'email': email, 'otp': otp},
      );
      final authResponse = AuthResponse.fromJson(response.data);
      if (authResponse.token != null) {
        await storageService.saveToken(authResponse.token!);
      }
      if (authResponse.user != null) {
        await storageService.saveUserData(jsonEncode(authResponse.user!.toJson()));
      }
      return authResponse;
    } catch (e) {
      throw ApiClient.getErrorMessage(e);
    }
  }

  Future<AuthResponse> forgotPassword(String email) async {
    try {
      final response = await apiClient.dio.post(
        '/api/v1/auth/forgot-password',
        data: {'email': email},
      );
      return AuthResponse.fromJson(response.data);
    } catch (e) {
      throw ApiClient.getErrorMessage(e);
    }
  }

  Future<AuthResponse> resendOtp(String email, String purpose) async {
    try {
      final response = await apiClient.dio.post(
        '/api/v1/auth/resend-otp',
        data: {'email': email, 'purpose': purpose},
      );
      return AuthResponse.fromJson(response.data);
    } catch (e) {
      throw ApiClient.getErrorMessage(e);
    }
  }

  Future<AuthResponse> resetPassword(String email, String otp, String newPassword) async {
    try {
      final response = await apiClient.dio.post(
        '/api/v1/auth/reset-password',
        data: {'email': email, 'otp': otp, 'new_password': newPassword},
      );
      return AuthResponse.fromJson(response.data);
    } catch (e) {
      throw ApiClient.getErrorMessage(e);
    }
  }

  Future<UserModel?> getProfile() async {
    try {
      final response = await apiClient.dio.get('/api/v1/auth/me');
      if (response.data != null && response.data['user'] != null) {
        final user = UserModel.fromJson(response.data['user']);
        await storageService.saveUserData(jsonEncode(user.toJson()));
        return user;
      }
      return null;
    } catch (e) {
      // Fallback to local stored user if offline
      final storedJson = await storageService.getUserData();
      if (storedJson != null) {
        return UserModel.fromJson(jsonDecode(storedJson));
      }
      return null;
    }
  }

  Future<void> logout() async {
    try {
      await apiClient.dio.post('/api/v1/auth/logout');
    } catch (_) {}
    await storageService.clearAll();
  }
}
